apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts

```diff
@@ -0,0 +1,584 @@
+import { escapeHtml, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+
+import { Modal } from '../../components/Modal.html';
+import { IdSchema } from '../../lib/db-types';
+import { CourseWithPermissionsSchema } from '../../models/course';
+
+export const SelectedAssessmentsSchema = z.object({
+  title: z.string(),
+  course_instance_id: IdSchema,
+  assessments: z.array(
+    z.object({
+      assessment_id: IdSchema,
+      color: z.string(),
+      label: z.string(),
+    }),
+  ),
+});
+type SelectedAssessments = z.infer<typeof SelectedAssessmentsSchema>;
+
+type CourseWithPermissions = z.infer<typeof CourseWithPermissionsSchema>;
+
+interface SharingSet {
+  id: string;
+  name: string;
+  in_set: string;
+}
+
+export function InstructorQuestionSettings({
+  resLocals,
+  questionTestPath,
+  questionTestCsrfToken,
+  isEnterprise,
+  questionGHLink,
+  qids,
+  a_with_q_for_all_ci,
+  sharingEnabled,
+  sharingSetsIn,
+  sharingSetsOther,
+  editableCourses,
+  infoPath,
+}: {
+  resLocals: Record<string, any>;
+  questionTestPath: string;
+  questionTestCsrfToken: string;
+  isEnterprise: boolean;
```

Comment:
Rather than passing this in from the route handler, consider just directly invoking `isEnterprise()` here.

Reply:
Removed call for `inEnterprise()` from the route handler and moved the call inline where it is used in the `PubliclyShareModal` in `instructorQuestionSettings.html.ts`.